src/palace/manager/celery/tasks/axis.py

```diff
@@ -214,53 +224,34 @@ def log_run_end_message() -> None:
             )
             log_run_end_message()
             return
-
         api = create_api(session=session, collection=collection)
         start_seconds = time.perf_counter()
-        total_imported_in_current_task = 0
-        while len(identifiers) > 0:
-            batch = identifiers[:batch_size]
-
+        batch = identifiers[:batch_size]
+        batch_length = len(batch)
+        if batch_length > 0:
             try:
                 for metadata, circulation in api.availability_by_title_ids(
                     title_ids=batch
                 ):
                     process_book(task, session, api, metadata, circulation)
-            except (ObjectDeletedError, StaleDataError) as e:
+            except (ObjectDeletedError, StaleDataError, OperationalError) as e:
```

Comment:
Ideally, I'd like to not do this. I think adding this here papers over what is a bigger problem.

We should be able to sort out a way not to have deadlocks happen, rather then retry when they do happen. Deadlocks are slow, they stop all the processing while the deadlock is happening, and they take awhile to time out. Its also possible to have deadlocks that the PG deadlock detector does not catch, that just freeze until the session times out.